src/App.vue

```diff
@@ -57,6 +58,19 @@ export default {
 	},
 	methods: {
 		routing(currentRoute) {
+			try {
+				if (loadState('tables', 'contextId', undefined)) {
+					// prepare route, when Context is opened from navigation bar
+					const contextId = loadState('tables', 'contextId', undefined);
+					const originalUrl = window.location.href
+					this.$router.replace('/application/' + contextId).catch(() => {})
+					// reverts turning /apps/tables/app/28 into /apps/tables/app/28#/application/28
```

Comment:
Seems like an OK approach for now.

We could also change the vue-router so that it uses the exact same routes as the backend defines, but that might require a few more changes to have the old urls still work.

https://github.com/nextcloud/deck/commit/9f98f0bd5fe85cd6e42688467c5bca9a75090fc4
https://github.com/nextcloud/deck/pull/1977/files

Reply:
interesting switch! Thanks for the pointer